products/views.py

```diff
@@ -1,3 +1,73 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http    import JsonResponse
+from django.views   import View
+
+from products.models import Categories, Images, Products, Sale, Review
+from orders.models   import Options
+
+class ProductListView(View):
+    def get(self,request):
+        category    = request.GET.get('name', None)
+        products    = Products.objects.all()
+
+        if category=="ourfoods":
+            treats = Categories.objects.get(name="treats")
+            products = Products.objects.exclude(category_id = treats)
+
+        else:
+            category_id    = Categories.objects.get(name=category).id
+            products    = Products.objects.filter(category_id=category_id)
+
+        products_list = [{
+        "id"            : products[i].id,
+        "name"          : products[i].name,
+        "main_image"    : Images.objects.filter(product_id = products[i].id).first().image_url,
+        "sub_image"     : Images.objects.filter(product_id = products[i].id).last().image_url,
+        "category"      : products[i].category.name,
+        "price"         : products[i].price,
+        "stock"         : products[i].stock_status,
+        "sale_rate"     : products[i].sale_rate.sale_rate} for i in range(len(products))]
+        return JsonResponse({"products_list":products_list}, status=200)
+
+class ProductDetailView(View):
+    def get(self,request, product_id):
+        try:
+            product = Products.objects.get(id=product_id)
+            option = Options.objects.all()
+            data = {
+                "id"            : product.id,
+                "name"          : product.name,
+                "main_image"    : Products.objects.get(id=product_id).images_set.all().first().image_url,
+                "sub_image"     : Products.objects.get(id=product_id).images_set.all().last().image_url,
+                "category"      : Categories.objects.get(id=product.category_id).name,
```

Comment:
연관된 테이블에 아래와 같이 인스턴스 속성에 접근해서 데이터를 가져올 수 있습니다!
main_image, sub_image 도 아래와 같이 product부터 접근해서 가져와주세요~
```suggestion
                "category"      : product.category.name,
```

Reply:
네 수정완료하였습니다.

Reply:
수정 완료 했습니다.